control-plane/pkg/prober/async_prober.go

```diff
@@ -161,3 +161,30 @@ func (a *asyncProber) Probe(ctx context.Context, addressable Addressable, expect
 func (a *asyncProber) enqueueArg(_ string, arg interface{}) {
 	a.enqueue(arg.(types.NamespacedName))
 }
+
+func (a *asyncProber) RotateRootCaCerts(caCerts *string) error {
+	newClient, err := makeHttpClientWithTLS(caCerts)
+	if err != nil {
+		return err
+	}
+
+	a.clientMu.Lock()
+	defer a.clientMu.Unlock()
+	a.client.(*http.Client).CloseIdleConnections()
+	a.client = newClient
+	return nil
+}
+
+func makeHttpClientWithTLS(caCerts *string) (*http.Client, error) {
+	tlsClient := eventingtls.ClientConfig{CACerts: caCerts}
+	tlsClientConfig, err := eventingtls.GetTLSClientConfig(tlsClient)
+	if err != nil {
+		return nil, err
+	}
+
+	return &http.Client{
+		Transport: &http.Transport{
```

Comment:
Instead of creating a new one, I'd close the default one: https://github.com/knative/eventing/blob/825a237c4255f18f3006d96bfa97accab39473f8/pkg/adapter/v2/cloudevents.go#L154-L155

The difference is that, it has some good default values that we're not setting here https://go.dev/src/net/http/transport.go:
```golang
// DefaultTransport is the default implementation of Transport and is
// used by DefaultClient. It establishes network connections as needed
// and caches them for reuse by subsequent calls. It uses HTTP proxies
// as directed by the environment variables HTTP_PROXY, HTTPS_PROXY
// and NO_PROXY (or the lowercase versions thereof).
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: defaultTransportDialContext(&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}),
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}
```